### PR TITLE
feat: Add filter pathdoesntexist in scm generator

### PR DESCRIPTION
**Pull request by rishabh625**
Fixes :- [#555](https://github.com/argoproj/applicationset/issues/555)
In a way can provide fix to [this](https://github.com/argoproj/argo-cd/issues/4808)
Repo with certain file if present in application will be ignored by applicationset
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by rishabh625** [comment]
@jannfis  @ishitasequeira @chetan-rns @crenshaw-dev : Can u please review?

File: applicationset/services/scm_provider/types.go [hunk 1]
```diff
@@ -24,11 +24,12 @@ type SCMProviderService interface {
...
 // A compiled version of SCMProviderGeneratorFilter for performance.
 type Filter struct {
-	RepositoryMatch *regexp.Regexp
-	PathsExist      []string
-	LabelMatch      *regexp.Regexp
-	BranchMatch     *regexp.Regexp
-	FilterType      FilterType
+	RepositoryMatch  *regexp.Regexp
+	PathsExist       []string
+	PathsDoesntExist []string
```
Comment (crenshaw-dev) [inline applicationset/services/scm_provider/types.go:29]: Can this be `PathsDoNotExist` instead?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
One nitpick, otherwise lgtm!

File: docs/operator-manual/applicationset/Generators-SCM-Provider.md [hunk 2]
```diff
@@ -189,15 +189,17 @@ spec:
...
       - repositoryMatch: ^otherapp
         pathsExist: [helm]
+        pathsDoesntExist: [disabledrepo.txt]
   template:
   # ...
 ```
 
 * `repositoryMatch`: A regexp matched against the repository name.
 * `pathsExist`: An array of paths within the repository that must exist. Can be a file or directory, but do not include the trailing `/` for directories.
+* `pathsDoesntExist`: An array of paths within the repository that must not exist. Can be a file or directory, but do not include the trailing `/` for directories.
```
Comment (ishitasequeira) [inline docs/operator-manual/applicationset/Generators-SCM-Provider.md:202]: Is it possible to have a check for trailing "/" and trim it while storing the value or checking for path instead?

**Review by ishitasequeira — CHANGES_REQUESTED** [review]
Just a minor feedback. rest lgtm!

File: applicationset/services/scm_provider/types.go (hunk 1 above)
Comment (ishitasequeira) [reply applicationset/services/scm_provider/types.go:29]: +1

**Review by crenshaw-dev — APPROVED** [review]
LGTM!

**Review by ishitasequeira — APPROVED** [review]
LGTM!

**Pull request merged**